prime-router/src/main/kotlin/Element.kt

```diff
@@ -420,7 +423,13 @@ data class Element(
                 try {
                     val optionalDateTime = variableDateTimePattern
                     val df = DateTimeFormatter.ofPattern(optionalDateTime)
-                    val ta = df.parseBest(formattedValue, OffsetDateTime::from, LocalDateTime::from, Instant::from)
+                    val ta = df.parseBest(
+                        formattedValue,
+                        OffsetDateTime::from,
+                        LocalDateTime::from,
+                        Instant::from,
+                        LocalDate::from
+                    )
```

Comment:
This makes our CSV date parsing even more flexible, and fixes some bugs with dates being passed in, which are all big wins. The key thing here is that if we expect a date time value and only get a date, we assume 00:00 and move on. I think that this is an acceptable choice.

Reply:
HARD AGREE